### PR TITLE
Pull Request for OpenSSL 3.0 support

### PR DESCRIPTION
**Pull request by LeftTry**

### CODE REVIEW

**Comment by vstakhov** [comment]
Thanks, it is now much more readable.

File: src/libserver/dkim.c [hunk 1]
```diff
@@ -155,8 +156,8 @@ struct rspamd_dkim_key_s {
 	gsize decoded_len;
 	char key_id[RSPAMD_DKIM_KEY_ID_LEN];
 	union {
-		RSA *key_rsa;
-		EC_KEY *key_ecdsa;
+		//RSA *key_rsa;
+		//EC_KEY *key_ecdsa;
```
Comment (vstakhov) [inline src/libserver/dkim.c:160]: We can remove the whole union I suppose to make the code clearer.

Comment (LeftTry) [reply src/libserver/dkim.c:160]: We should remove whole code that is connected to `RSA` and `ECDSA` or move it to the #if #endif section?

Comment (vstakhov) [reply src/libserver/dkim.c:160]: I think that using EVP methods is much better. It should work with OSSL 1.x as well from what I know. We will break compatibility with OSSL < 1.0 but nobody should care about that crap.

Comment (LeftTry) [reply src/libserver/dkim.c:160]: Ok. I'll remove the whole support. Also, should we leave ed25519 in the union or remove it?

Comment (vstakhov) [reply src/libserver/dkim.c:160]: In fact, I will iterate on this PR by myself, it seems we can clean up the code massively.

**Comment by vstakhov** [comment]
It seems there are many formatting changes that are not intended. Please use `clang-format` with the settings that are shipped with Rspamd: https://github.com/rspamd/rspamd/blob/master/.clang-format

**Comment by vstakhov** [comment]
I have merged it occasionally. Git is such a messy thing when it comes to multiple remotes...

**Pull request merged**